File: starterAIs/Starter.rb
STDOUT.sync = true # DO NOT REMOVE

class Cell
  attr_accessor :index
  attr_accessor :type
  attr_accessor :resources
  attr_accessor :neighbors
  attr_accessor :my_ants
  attr_accessor :opp_ants

  def initialize(index, type, resources, neighbors, my_ants, opp_ants)
      @index = index
      @type = type
      @resources = resources
      @neighbors = neighbors
      @my_ants = my_ants
      @opp_ants = opp_ants
  end
end

cells = []

number_of_cells = gets.to_i # amount of hexagonal cells in this map
number_of_cells.times do |i|
  inputs = gets.split.map &:to_i
  _type = inputs[0] # 0 for empty, 1 for eggs, 2 for crystal
  initial_resources = inputs[1] # the initial amount of eggs/crystals on this cell
  neigh_0 = inputs[2] # the index of the neighbouring cell for each direction
  neigh_1 = inputs[3]
  neigh_2 = inputs[4]
  neigh_3 = inputs[5]
  neigh_4 = inputs[6]
  neigh_5 = inputs[7]
  cell = Cell.new(
    i,
    _type,
    initial_resources,
    [neigh_0, neigh_1, neigh_2, neigh_3, neigh_4, neigh_5].select { |id| id > -1 },
    0,
    0
  )
  cells.append cell
end
number_of_bases = gets.to_i
inputs = gets.split
my_bases = []
for i in 0..(number_of_bases-1)
  my_base_index = inputs[i].to_i
  my_bases.append my_base_index
end
inputs = gets.split
opp_bases = []
for i in 0..(number_of_bases-1)
  opp_base_index = inputs[i].to_i
  opp_bases.append opp_base_index
end

# game loop
loop do
  number_of_cells.times do |i|
    inputs = gets.split.map &:to_i
    resources = inputs[0] # the current amount of eggs/crystals on this cell
    my_ants = inputs[1] # the amount of your ants on this cell
    opp_ants = inputs[2] # the amount of opponent ants on this cell

    cells[i].resources = resources
    cells[i].my_ants = my_ants
    cells[i].opp_ants = opp_ants
  end

  # WAIT | LINE <sourceIdx> <targetIdx> <strength> | BEACON <cellIdx> <strength> | MESSAGE <text>
  actions = []

  # TODO: choose actions to perform and push them into actions
  # To debug: STDERR.puts "Debug messages..."
  if actions.length == 0
    puts 'WAIT'
  else
    puts actions.join ";"
  end
end

File: starterAIs/Starter.js
class Cell {
  constructor(index, type, resources, neighbors, myAnts, oppAnts) {
      this.index = index
      this.type = type
      this.resources = resources
      this.neighbors = neighbors
      this.myAnts = myAnts
      this.oppAnts = oppAnts
  }
}

const cells = []

const numberOfCells = parseInt(readline()); // amount of hexagonal cells in this map
for (let i = 0; i < numberOfCells; i++) {
  const inputs = readline().split(' ');
  const type = parseInt(inputs[0]); // 0 for empty, 1 for eggs, 2 for food
  const initialResources = parseInt(inputs[1]); // the initial amount of eggs/crystals on this cell
  const neigh0 = parseInt(inputs[2]); // the index of the neighbouring cell for each direction
  const neigh1 = parseInt(inputs[3]);
  const neigh2 = parseInt(inputs[4]);
  const neigh3 = parseInt(inputs[5]);
  const neigh4 = parseInt(inputs[6]);
  const neigh5 = parseInt(inputs[7]);

  const cell = new Cell(
      i,
      type,
      initialResources,
      [neigh0, neigh1, neigh2, neigh3, neigh4, neigh5].filter(id => id > -1),
      0,
      0
  )
  cells.push(cell)
}

const numberOfBases = parseInt(readline());
const myBases = readline().split(' ').map(n => parseInt(n))
const oppBases = readline().split(' ').map(n => parseInt(n))

// game loop
while (true) {
  for (let i = 0; i < numberOfCells; i++) {
      const inputs = readline().split(' ')
      const resources = parseInt(inputs[0]); // the current amount of eggs/crystals on this cell
      const myAnts = parseInt(inputs[1]); // the amount of your ants on this cell
      const oppAnts = parseInt(inputs[2]); // the amount of opponent ants on this cell

      cells[i].resources = resources
      cells[i].myAnts = myAnts
      cells[i].oppAnts = oppAnts
  }


  // WAIT | LINE <sourceIdx> <targetIdx> <strength> | BEACON <cellIdx> <strength> | MESSAGE <text>
  const actions = []

  // TODO: choose actions to perform and push them into actions. E.g:
  for (const cell of cells) {
      if (cell.resources > 0) {
          actions.push(`LINE ${myBases[0]} ${cell.index} 1`)
          break
      }
  }

  // To debug: console.error('Debug messages...');
  if (actions.length === 0) {
      console.log('WAIT');
  } else {
      console.log(actions.join(';'))
  }
}
